[UT] Fix unstable primary key backup FEUT

srlch opened the pull request:
Why I'm doing:
primary key backup FEUT will failed sometime.The problem is BackupJobMaterializedViewTest and
BackupJobPrimaryKeyTest use the same root meta data snapshot path. And it may be deleted in
BackupJobMaterializedViewTest'end() when BackupJobPrimaryKeyTest is running. BackupJobPrimaryKeyTest
will get no file and dictoinary exception sometime.

What I'm doing:
Save all paths which needed to be deleted in all sub UT. And delete the paths in the end() function.
Do not delete the root path.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [x] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

imay commented on fe/fe-core/src/main/java/com/starrocks/backup/BackupJob.java:
```diff
@@ -358,6 +358,21 @@ public synchronized void run() {
         }
+
+        // just use for FEUT, clear the path
+        if (testPrimaryKey && state == BackupJobState.FINISHED && localJobDirPath != null) {
```
I don't think it is appropriate to change the original code just for UT.
Why not delete this path in the UT? That seems a better way than this.

srlch replied:
done, has fixed the code

srlch committed:
Why I'm doing:
primary key backup FEUT will failed sometime.The problem is BackupJobMaterializedViewTest and
BackupJobPrimaryKeyTest use the same root meta data snapshot path. And it may be deleted in
BackupJobMaterializedViewTest'end() when BackupJobPrimaryKeyTest is running. BackupJobPrimaryKeyTest
will get no file and dictoinary exception sometime.

What I'm doing:
Save all paths which needed to be deleted in all sub UT. And delete the paths in the end() function.
Do not delete the root path.

Signed-off-by: srlch <linzichao@starrocks.com>

srlch commented:
ignore backport check: 3.2.4
